### PR TITLE
Feature events localhuman

### PR DESCRIPTION
**Pull request by localhuman**
- Added `execution_success` to `SmartContractEvent`, so that `Notify` events can have this context for when they are being processed
- Moved dispatch of `Notify` to occur after execution of the contract, and include the `execution_success` in the event

Notes: Ideally, when dispatching the events we'd be passing around `UInt160` or `UInt256` for tx hash or for contract hashes, that way any service consuming the events will have a flexible data type to use

### CODE REVIEW

File: neo/EventHub.py [hunk 1]
```diff
@@ -31,39 +42,72 @@ class SmartContractEvent:
...
-@events.on("*")
-@events.on("*.*")
-def on_any_event(*args):
-    print("")
-    print("=EVENT: %s" % args)
-    print("")
+
+
+@events.on(SmartContractEvent.RUNTIME_LOG)
+def on_sc_log(*args):
```
Comment (metachris) [inline neo/EventHub.py]: if you know these events always include a SmartContractEvent as argument, we could change the method signature from this:

    def on_sc_log(*args):
        if len(args) > 0 and type(args[0]) is SmartContractEvent:
            sc_event = args[0]

to

    def on_sc_log(sc_event):

File: neo/SmartContract/StateMachine.py [hunk 2]
```diff
@@ -85,6 +83,42 @@ def CheckStorageContext(self, context):
...
+        if success:
+            self.Commit()
+
+        height = Blockchain.Default().Height
+        tx_hash = engine.ScriptContainer.Hash
+
+        # dispatch all notify events, along with the success of the contract execution
+        for notify_event_args in self.notifications:
+
+            dispatch_smart_contract_event(SmartContractEvent.RUNTIME_NOTIFY, notify_event_args.State, notify_event_args.ScriptHash.ToString(), height, tx_hash.ToString(), success)
```
Comment (metachris) [inline neo/SmartContract/StateMachine.py]: What data type is `notify_event_args.State` here? Would be great for usability if Runtime.Notify events always have a list of python objects as payload.

File: neo/VM/InteropService.py [hunk 3]
```diff
@@ -357,5 +355,11 @@ def stack_item_to_py(stack_item):
     elif isinstance(stack_item, ByteArray):
         return stack_item.GetBigInteger()
 
+    elif isinstance(stack_item, InteropInterface):
+        return stack_item.GetInterface()
+
+    elif isinstance(stack_item, Struct):
+        return stack_item.GetArray()
```
Comment (metachris) [inline neo/VM/InteropService.py]: Not sure, but should we convert the array items like for `Array` above?

    return [stack_item_to_py(item) for item in stack_item.GetArray()]

**Comment by metachris** [comment]
Commented at a few points inline. Great work, love it! 👍

**Comment by metachris** [comment]
PS: Also I'm all for your suggestion re "Ideally, when dispatching the events we'd be passing around UInt160 or UInt256 for tx hash or for contract hashes"

File: neo/VM/InteropService.py (hunk 3 above)
Comment (localhuman) [reply neo/VM/InteropService.py]: I haven't done any testing with `Struct` yet, but that should work

**Pull request merged**